### PR TITLE
Weekend-only mode for satellite auto-updates

### PR DESCRIPTION
**Pull request by brandonSc**
Users can specify `--maintenance-weekends-only` as a launch flag to ensure their satellites only auto-update over the weekend. They can also retroactively enable this using the `satellite update` command.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/docker/distribution v2.8.1+incompatible
 	github.com/docker/go-connections v0.4.0
 	github.com/dustin/go-humanize v1.0.0
-	github.com/earthly/cloud-api v1.0.1-0.20230216165236-281e4a60a762
+	github.com/earthly/cloud-api v1.0.1-0.20230217185922-87efbca903da
```
Comment (brandonSc) [inline go.mod:18]: Need to update this once the backend change merges.

**Review by vladaionescu — APPROVED** [review]
Add an entry in the changelog too?

**Review by alexcb — APPROVED** [review]

File: cloud/satellite.go [hunk 2]
```diff
@@ -120,15 +122,27 @@ func (c *Client) DeleteSatellite(ctx context.Context, name, orgID string) error
...
+	OrgID                   string
+	Size                    string
+	Platform                string
+	PinnedVersion           string
+	MaintenanceWindowStart  string
+	MaintenanceWeekendsOnly bool
+	FeatureFlags            []string
+}
+
+func (c *Client) LaunchSatellite(ctx context.Context, opt LaunchSatelliteOpt) error {
```
Comment (nelsam) [inline cloud/satellite.go:136]: Stylistic nit: I know they're a lot more effort to implement, but I really prefer [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) over the option struct pattern. They are harder to write, but easier to comprehend and use.

**Review by nelsam — APPROVED** [review]
lgtm; minor comment about functional options, but that's not critically important to this PR.

File: cloud/satellite.go (hunk 2 above)
Comment (brandonSc) [reply cloud/satellite.go:136]: Cool, I can try that out in a future PR, since I want to get his merged asap for 0.7 release

File: go.mod (hunk 1 above)
Comment (brandonSc) [reply go.mod]: Done. Backend is merged and this is up to date now.

**Pull request merged**